### PR TITLE
Add macOS to CI matrix builds

### PR DESCRIPTION
**Pull request by robert-scheck**
This hopefully avoids issues like https://github.com/rpki-client/rpki-client-portable/issues/84 in the future.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -10,20 +10,38 @@ on:
...
           CC: ${{ matrix.compiler }}
-      - name: install missing deps
+      - name: install missing Linux deps
         run: sudo apt-get install libtls-dev
+        if: runner.os == 'Linux'
+      - name: install missing macOS deps
+        run: |
+          brew install libretls openssl@3 rsync
+          echo "LDFLAGS=-L/usr/local/opt/openssl@3/lib" >> $GITHUB_ENV
+          echo "CPPFLAGS=-I/usr/local/opt/openssl@3/include" >> $GITHUB_ENV
```
Comment (carlocab) [inline .github/workflows/build.yaml:43]: You can also do this if you don't want to hardcode the prefix.
```suggestion
          echo "LDFLAGS=-L$(brew --prefix openssl@3)/lib" >> $GITHUB_ENV
          echo "CPPFLAGS=-I$(brew --prefix openssl@3)/include" >> $GITHUB_ENV
```

**Review by carlocab — APPROVED** [review]
LGTM

File: .github/workflows/build.yaml (hunk 1 above)
Comment (robert-scheck) [reply .github/workflows/build.yaml]: For consistency, this makes absolutely sense – applied.

**Comment by cjeker** [comment]
Merged this and did some further cleanup in the compiler matrix.
There is little point in testing older compilers since we're interested in new warnings and errors.
Also only do the test run on linux systems.

**Pull request merged**